src/breakpoints/body.tsx

```diff
@@ -22,26 +21,52 @@ export class Body extends ReactWidget {
 }
 
 const BreakpointsComponent = ({ model }: { model: Breakpoints.Model }) => {
-  const [breakpoints, setBreakpoints] = useState(model.breakpoints);
+  const [breakpoints, setBreakpoints] = useState(
+    Array.from(model.breakpoints.entries())
+  );
 
   useEffect(() => {
     const updateBreakpoints = (
       _: Breakpoints.Model,
       updates: Breakpoints.IBreakpoint[]
     ) => {
-      if (ArrayExt.shallowEqual(breakpoints, updates)) {
-        return;
-      }
-      setBreakpoints(updates);
+      setBreakpoints(Array.from(model.breakpoints.entries()));
+    };
+
+    const restoreBreakpoints = (_: Breakpoints.Model) => {
+      setBreakpoints(Array.from(model.breakpoints.entries()));
```

Comment:
This looks like the `restored` and `changed` update logic could be merged into one?

Reply:
This depends whether we want to keep the current logic, where we ask for all the breakpoints each time there is an update instead of emitting the breakpoints that changed in the signal.
In the first case, we can change the signal signature and merge the `restored` and `update` functions.

Reply:
How about we leave it like this and open a new issue, so we can have a look again later? (if needed)